packml_stats_loader/src/packml_stats_loader.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2019, PlusOne Robotics
+ * All rights reserved.
+*/
+
+#include "packml_stats_loader/packml_stats_loader.h"
+
+namespace packml_stats_loader
+{
+  PackmlStatsLoader::PackmlStatsLoader(const ros::NodeHandle &pnh): pnh_(pnh)
+  {
+
+    // Get parameters
+    if (!pnh_.getParam("packml_stats_location", packml_stats_location_))
+    {
+      ROS_FATAL_STREAM("Missing param: packml_stats_location. Unable to load or save stats. Shutting down.");
+      ros::shutdown();
+    }
+    bool load_packml_stats = false;
+    if (!pnh_.getParam("load_packml_stats", load_packml_stats))
+    {
+      ROS_WARN_STREAM("Missing param: load_packml_stats. Defaulting to false.");
+    }
+    float save_stats_rate = 1.0;
+    if (!pnh_.getParam("save_stats_rate", save_stats_rate))
+    {
+      ROS_WARN_STREAM("Missing param: save_stats_rate. Defaulting to 1.0Hz.");
+    }
+
+    // Load stats
+    if (load_packml_stats)
+    {
+      ros::ServiceClient load_stats_client = pnh_.serviceClient<packml_msgs::LoadStats>("load_stats");
+      ros::service::waitForService(load_stats_client.getService());
+      packml_msgs::LoadStats load_stats_srv;
+      load_stats_srv.request.stats = loadStats();
+
+      if (!load_stats_client.call(load_stats_srv))
+      {
+        ROS_ERROR_STREAM("Failed to call service " << load_stats_client.getService());
```

Comment:
Should this be a FATAL condition like above? Presumably if you're expecting a continuation of stats it might be confusing if the load fails.

Reply:
Good call, changed to fatal